### ISSUE TITLE
Regression - escape character was not considered when comparing column names

### ISSUE THREAD

**Issue by moming00** [ClickHouse/clickhouse-go#828]
### Issue description
When upgrade ch-go from v2.3.0 to 2.4.0, our app is no longer working because of the following error:

"error":"clickhouse.Conn.PrepareBatch INSERT INTO `default`.`test_ck_query` (`timestamp`,`dim00`,`dim01`,`dim02`,`dim03`,`metric00`,`metric01`,`metric02`,`metric03`): block cannot be sorted - missing columns in requested order: [timestamp dim00 dim01 dim02 dim03 metric00 metric01 metric02 metric03]"

![image](https://user-images.githubusercontent.com/73276360/203243527-7adeac3f-c3c0-4fc4-bc22-90c9779899db.png)

possibility because of the SortColumns code in block.go did not consider the escape chars.

**Comment by gingerwizard** [comment]
Reproducible
```
func Test828(t *testing.T) {
	var (
		conn, err = clickhouse_tests.GetConnection("issues", clickhouse.Settings{
			"max_execution_time": 60,
			"flatten_nested":     0,
		}, nil, &clickhouse.Compression{
			Method: clickhouse.CompressionLZ4,
		})
	)
	ctx := context.Background()
	require.NoError(t, err)
	env, err := clickhouse_tests.GetTestEnvironment(testSet)
	require.NoError(t, err)
	ddl := fmt.Sprintf("CREATE TABLE `%s`.`test_828` (Col1 String, Col2 UInt8) Engine MergeTree() ORDER BY tuple()", env.Database)
	defer func() {
		conn.Exec(ctx, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`test_828`", env.Database))
	}()
	conn.Exec(ctx, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`test_828`", env.Database))
	require.NoError(t, conn.Exec(ctx, ddl))
	batch, err := conn.PrepareBatch(ctx, fmt.Sprintf("INSERT INTO `%s`.`test_828` (`Col1`, `Col2`)", env.Database))
	require.NoError(t, err)
	require.NoError(t, batch.Append(
		"Clicky McClickHouse",
		uint8(1),
	))
	require.NoError(t, batch.Send())
}
```

Fixing and with release 2.4.1

**Comment by gingerwizard** [comment]
Will be closed by https://github.com/ClickHouse/clickhouse-go/pull/829. 2.4.1 will be available once merge @moming00

**Comment by moming00** [comment]
> Will be closed by #829. 2.4.1 will be available once merge @moming00

thanks for the quick action =)
